frontend/src/components/managePermissions/table/DataTableHeader.vue

```diff
@@ -0,0 +1,169 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+import InputText from 'primevue/inputtext';
+import router from '@/router';
+import { IconSize } from '@/enum/IconEnum';
+import { selectedApplicationDisplayText } from '@/store/ApplicationState';
+
+const props = defineProps({
+    btnLabel: {
+        type: String,
+        required: true,
+    },
+    btnRoute: {
+        type: String,
+        required: true
+    },
+    filter: {
+        type: String,
+        required: true
+    }
+});
+const emit = defineEmits(['change']);
+
+const proxyValue = computed({
+    get() {
+        return props.filter
+    },
+    set(newValue) {
+        emit('change', newValue)
+    }
+})
+
+</script>
+
+<template>
+    <div class="custom-data-table-header">
+        <h3>{{ selectedApplicationDisplayText }} users</h3>
+        <span>
+            This table shows all the users in
+            {{ selectedApplicationDisplayText }} and their permissions
+            levels
+        </span>
+    </div>
+
+    <div class="search-container">
+        <Button
+            class="btn-add-user"
+            :label="props.btnLabel"
+            @click="router.push(props.btnRoute)"
+        >
+            <Icon icon="add" :size="IconSize.small" />
+        </Button>
+        <span class="p-input-icon-left">
+            <Icon icon="search" :size="IconSize.small" />
+            <InputText
+                id="dashboardSearch"
+                class="dash-search"
+                placeholder="Search by keyword"
+                v-model="proxyValue"
+                :value="props.filter"
+            />
+        </span>
+    </div>
+</template>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/base.scss';
+
+.custom-data-table-header {
+    padding: 1rem 1rem 1.5rem;
+    background-color: $light-layer-two;
+    h3 {
+        @extend %heading-03;
+        margin: 0;
+        padding: 0;
+    }
+
+    span {
+        @extend %body-compact-01;
+        margin: 0;
+        padding: 0;
+        color: $light-text-secondary;
+    }
+}
+
+.search-container {
+    display: flex;
+}
+
+.btn-add-user {
+    width: 16rem;
+    z-index: 2;
+    border-radius: 0rem;
+}
+
+.dash-search {
+    border-radius: 0 0 0 0;
+}
+
+.btn-icon {
```

Comment:
this class I'm not seeing in use in this file, also the .btn-icon:disabled

Reply:
just removed it